### PR TITLE
Remove .commands

### PR DESCRIPTION
**Pull request by TheKodeToad**
Since this seemed to be what a few people wanted?

### CODE REVIEW

File: src/handlers/event/delete_on_reaction.rs [hunk 1]
```diff
@@ -14,10 +14,7 @@ pub async fn handle(ctx: &Context, reaction: &Reaction) -> Result<()> {
 		.wrap_err("Couldn't fetch message from reaction!")?;
 
 	if let Some(interaction) = &message.interaction {
-		if interaction.kind == InteractionType::Command
```
Comment (TheKodeToad) [inline src/handlers/event/delete_on_reaction.rs:17]: I removed this because I don't think you can have any other interaction kind on a message

Comment (getchoo) [reply src/handlers/event/delete_on_reaction.rs:17]: this is enum is marked as non-exhaustive and therefore *could* in the future. this is a purposeful guard as we should only be deleting messages from commands anyways

File: src/commands/general/delete_interaction.rs
Comment (getchoo) [inline src/commands/general/delete_interaction.rs:1]: we should also make sure this reaction is of type Command as well, for the reasons described below

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by getchoo** [comment]
we also might want to just remove the MessageAdd path for deleting responses. it would be a breaking change from nodefraction, but i think a pretty welcome one as messages being deleted with the X isn't actually documented anywhere and using the context menu is similar effort to reacting anyways. this would also save us a bit of time on processing each message, which is a plus

File: src/handlers/event/delete_on_reaction.rs (hunk 1 above)
Comment (TheKodeToad) [reply src/handlers/event/delete_on_reaction.rs:17]: Oh I guess starting an activity might be a different type and you wouldn't want people to hide that

**Comment by TheKodeToad** [comment]
Also added Manage Messages check to delete any message because why not

File: src/commands/general/delete_interaction.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+use eyre::Result;
```
Comment (getchoo) [inline src/commands/general/delete_interaction.rs:1]: (on mobile so no actual suggestion :trollface: ) as we're giving an explicit error variant to Result with `crate::Error` this isn't needed

File: src/commands/general/delete_interaction.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	}
+
+	let can_manage = interaction
+		.member
+		.as_ref()
+		.and_then(|m| m.permissions)
+		.and_then(|p| p.contains(Permissions::MANAGE_MESSAGES).then_some(true))
+		.is_some();
+
+	if !(interaction.user.id == ctx.author().id || can_manage) {
```
Comment (getchoo) [inline src/commands/general/delete_interaction.rs:27]: `if (interaction.user.id != ctx.author()) && !can_manage` is a bit more readable imo

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by TheKodeToad** [comment]
I think people use them too often to be removed

**Pull request closed without merging**